charts/backstage/values.yaml

```diff
@@ -166,3 +166,68 @@ networkPolicy:
     ##           matchLabels:
     ##             label: example
     customRules: []
+
+
+# -- PostgreSQL [chart configuration](https://github.com/bitnami/charts/blob/master/bitnami/postgresql/values.yaml)
+postgresql:
+
+  # -- Switch to enable or disable the PostgreSQL helm chart
+  enabled: true
+
+  # -- The authentication details of the Postgres database
+  auth:
+
+    # -- Name for a custom user to create
+    username: bn_backstage
+
+    # -- Password for the custom user to create
+    password: ""
+
+    # -- Name of existing secret to use for PostgreSQL credentials
+    existingSecret: ""
+
+    # -- The secret keys Postgres will look for to retrieve the relevant password
+    secretKeys:
+
+      # -- The key in which Postgres well look for, for the admin password, in the existing Secret
+      adminPasswordKey: admin-password
+
+      # -- The key in which Postgres well look for, for the user password, in the existing Secret
+      userPasswordKey: user-password
+
+      # -- The key in which Postgres well look for, for the replication password, in the existing Secret
+      replicationPasswordKey: replication-password
+  
+  # -- PostgreSQL architecture (`standalone` or `replication`)
+  architecture: standalone
+
+
+# -- External database configuration
+externalDatabase:
+
+  # -- Switch to enable or disable the externalDatabase connection
+  enabled: false
```

Comment:
do we need this flag? 🤔 
I'm thinking that `postgresql.enabled` should be enough, right?
Perhaps we could remove the entire `externalDatabase` block, since it's already possible to setup an external database by using one of the methods described in the README

Reply:
Yes I was debating on this, the reason I added the `externalDatabase` block was the make it a bit more declarative and purpose built. It also gives less ways for users to misconfigure or miss something as it will directly map onto the override variable comment you made [here](https://github.com/vinzscam/backstage-chart/pull/2#discussion_r900042672). As with the `extraEnvVars`, we rely on the users having correctly setup their env vars properly in the `app-config`'s. With the `externalDatabase` block, because it's used in the helper functions in the `.tpl` file, it directly will be mapped into the env var override - giving less chance of user error. I've got a bunch of `README.md` stuff on this as well which will explain quite clearly what's needed.

Reply:
but you could have `externalDatabase.enabled: true` and `postgresql.enabled: true` and in that case  `externalDatabase.enabled` is going to be ignored, which could be a bit misleading.

I feel the entire `externalDatabase` section could be removed, since backstage already provide these capabilities by providing the proper values through `app-config` or environment variables, as described in the README.

Reply:
Going to have a stab at this now. I get what you are referencing with the `extraEnvVarsSecrets` value. I'll try and remove the `externalDatabase` bit. The only possible UX pain point would be that we'd have to make it explicit that there will be two ways of configuring the database. For postgres, you can just specify the secret name in the values file, or let the Chart generate the secret for you and it will just automatically wire everything up. But for the externalDatabase, you'll have to provide it as a separate secret and reference that secret in the `extraEnvVarsSecrets` list. This will then override the environment variables that will be provided in the deployment yaml.

Reply:
Also, for the configuring of an `externalDatabase` When a user provides the `POSTGRES_HOST` variable via the `extraEnvVars`, it will just end up being fed into the deployment as an extra env, as opposed to being used here: https://github.com/ChrisJBurns/backstage-chart/blob/main/charts/backstage/templates/backstage-deployment.yaml#L75

Reply:
>I feel the entire externalDatabase section could be removed, since backstage already provide these capabilities by providing the proper values through app-config or environment variables, as described in the README.

Also, it's worth noting that currently the externalDatabase config is being fed as environment variables here  https://github.com/ChrisJBurns/backstage-chart/blob/main/charts/backstage/templates/backstage-deployment.yaml#L75. So whether its the internal chart postgres or the external database, it still goes through the environment variable.

Reply:
Have sketched some code and this is what it would be like with both approaches:

Without `externalDatabase` block:

Values.yaml
```yaml
backstage.extraEnvVars:
 - name: POSTGRES_HOST
    value: database.host.com
 - name: POSTGRES_PORT
    value: 5432
 - name: POSTGRES_USER     
    value: username
backstage.extraEnvVarsSecrets:
 - backstage-secrets
```

This will then get fed into the Deployment as extraEnvVars in addition to the `POSTGRES_*` variables already in the Deployment manifest. Additionally, if someone leaves `postgres.enabled: true` on accident you'll have both sets of `POSTGRES_*` variables being provided into the deployment - which will cause funky behaviour.

With `externalDatabase` config block in values.yaml

```yaml
externalDatabase:
  enabled: true
  config:
    host: "database.com"
    port: "5432"
    auth:
      username: "username"
      existingSecret: "backstage-secret"
      existingSecretPasswordKey: "POSTGRES_PASSWORD" # this will take the value from the already created secret
```

With the above, all of the values for either the `postgres` or external postgres - with help from the helper functions - will all get put into the environment variables that are already existing in the deployment. I can even add an etra check in there to error if both `postgres.enabled: true` and `externalDatabase.enabled: true` To avoid people making errors.

Hopefully I've explained it better with the examples above. It's pretty much the same outcome with what you said about providing them as environment variables, it just saves having to write more code and use the `extraEnvVarSecrets` and `extraEnvVars` to inject additional `POSTGRES_*` variables into the deployment manifest, when they're already there.